src/vl-map-action.js

```diff
@@ -80,30 +117,38 @@ export class VlMapAction extends vlElement(HTMLElement) {
    */
   actionChanged() {
     const event = new Event(VlMapAction.NEW_ACTION_EVENT_NAME);
-    this.parentElement.dispatchEvent(event);
+    this._mapElement.dispatchEvent(event);
   }
 
   _layerChangedCallback() {
-    this._computeAction(this._map, this.layer);
+    this._mapElement.ready.then(() => {
+      if (this.layer) {
+        const action = this._createAction(this.layer);
+        if (action) {
+          this._mapElement.addAction(action);
+          this.actionChanged();
+          if (this._active) {
+            action.activate();
+          }
+          this._action = action;
+        }
+      }
```

Comment:
dit misschien ook even extracten naar een aparte functie met een beschrijvende naam

Reply:
De code zat vroeger in een `_computeAction` function, maar dat was voor mij niet echt beschrijvend waardoor ik deze functie vervangen heb door de code omdat ik vond dat deze goed genoeg weergeeft wat er gebeurt:
- `createAction`
- `addAction`
- `activate` indien active
- private `_action` setter

Uiteraard bereid om aparte functies te voorzien als het niet leesbaar genoeg is. Welke onderdelen zou je graag in functies willen? Opnieuw de `_computeAction` introduceren?